aya/src/maps/sock/sock_hash.rs

```diff
@@ -103,50 +101,34 @@ impl<T: Deref<Target = Map>, K: Pod> SockHash<T, K> {
     /// An iterator visiting all keys in arbitrary order. The iterator element
     /// type is `Result<K, MapError>`.
     pub fn keys(&self) -> MapKeys<'_, K> {
-        MapKeys::new(&self.inner)
+        MapKeys::new(self.inner.as_ref())
+    }
+
+    /// Returns the map's file descriptor, used for instances where programs
```

Comment:
Ha now I finally have something to link to! The documentation should follow this format https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components

something like:

```
Returns the map's file descriptor.

The returned file descriptor can be used to attach programs that work with socket maps, like [`SkMsg`] and [`SkSkb`].
```

Reply:
Ahh This is a good tidbit, thanks!